src/cljs/org/broadinstitute/firecloud_ui/utils.cljs

```diff
@@ -156,12 +132,19 @@
             (.send xhr data)
             (.send xhr)))))))
 
+
 (defn ajax-orch [path arg-map & {:keys [service-prefix] :or {service-prefix "/api"}}]
   (assert (= (subs path 0 1) "/") (str "Path must start with '/': " path))
-  (ajax (assoc
-         arg-map :url (str (config/api-url-root) service-prefix path)
-         :headers (merge {"Authorization" (str "Bearer " @access-token)}
-                         (:headers arg-map)))))
+  (let [on-done (:on-done arg-map)]
+    (ajax (assoc
+           arg-map :url (str (config/api-url-root) service-prefix path)
+           :headers (merge {"Authorization" (str "Bearer " @access-token)}
+                           (:headers arg-map))
+           :on-done (fn [{:keys [status-code] :as m}]
+                      ;; Handle auth token expiration
+                      (if (= status-code 401)
```

Comment:
this moves the check for 401s from ajax to ajax-orch. There are still two places that call ajax directly (for /config.json and /me), and they look safe. But maybe add a comment to the defn ajax to ensure nobody else unintentionally calls it directly?